File: backend/src/main/java/com/example/post/entity/Post.java
package com.example.post.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.web.bind.annotation.RequestMapping;

import java.sql.Date;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@Entity
@Table(name = "posts")
@NoArgsConstructor
@RequestMapping("/posts")
@AllArgsConstructor
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    @Transient
    private Long user_id;
    @Column(nullable = false)
    private String header;

    @Column(nullable = false)
    private String text;
    private Date date;



}

File: front/src/app/entity/post.ts
import {User} from "./user";

export class Post {
  id: number;
  user: User;
  header: string;
  text: string;
  date: string;


  constructor(id: number, user: User, header: string, text: string, date: string) {
    this.id = id;
    this.user = user;
    this.header = header;
    this.text = text;
    this.date = date;
  }
}
